File: php-clientlib-1.x/v1.x/examples/GetAllAdClients.php
<?php
/*
 * Copyright 2014 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * This example gets all ad clients for the specified account.
 *
 * Tags: accounts.adclients.list
 */
class GetAllAdClients {
  /**
   * Gets all ad clients for the specified account.
   *
   * @param $service Google_Service_AdSense AdSense service object on which to
   *     run the requests.
   * @param $accountId string the ID for the account to be used.
   * @param $maxPageSize int the maximum page size to retrieve.
   * @return array the last page of retrieved ad clients.
   */
  public static function run($service, $accountId, $maxPageSize) {
    $separator = str_repeat('=', 80) . "\n";
    print $separator;
    printf("Listing all ad clients for account %s\n", $accountId);
    print $separator;

    $optParams['maxResults'] = $maxPageSize;

    $pageToken = null;
    $adClients = null;
    do {
      $optParams['pageToken'] = $pageToken;
      $result = $service->accounts_adclients->listAccountsAdclients($accountId,
          $optParams);
      if (!empty($result['items'])) {
        $adClients = $result['items'];
        foreach ($adClients as $adClient) {
          printf("Ad client for product \"%s\" with ID \"%s\" was found.\n",
              $adClient['productCode'], $adClient['id']);
        }
        if (isset($result['nextPageToken'])) {
          $pageToken = $result['nextPageToken'];
        }
      } else {
        print "No ad clients found.\n";
      }
    } while ($pageToken);
    print "\n";

    return $adClients;
  }
}

File: v2/java/src/main/java/com/google/api/services/samples/adsense/cmdline/GetAllAdClients.java
/*
 * Copyright (c) 2021 Google LLC.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations under
 * the License.
 */

package com.google.api.services.samples.adsense.cmdline;

import com.google.api.services.adsense.v2.Adsense;
import com.google.api.services.adsense.v2.model.AdClient;
import com.google.api.services.adsense.v2.model.ListAdClientsResponse;
import java.util.ArrayList;
import java.util.List;

/**
*
* This example gets all ad clients for the logged in user's default account.
*
* Tags: accounts.adclients.list
*
*/
public class GetAllAdClients {

  /**
   * Runs this sample.
   *
   * @param adsense AdSense service object on which to run the requests.
   * @param accountId the ID for the account to be used.
   * @param maxPageSize the maximum page size to retrieve.
   * @throws Exception
   */
  public static List<AdClient> run(Adsense adsense, String accountId, int maxPageSize)
      throws Exception {
    System.out.println("=================================================================");
    System.out.printf("Listing all ad clients for account %s\n", accountId);
    System.out.println("=================================================================");

    // Retrieve ad client list in pages and display data as we receive it.
    List<AdClient> allAdClients = new ArrayList<AdClient>();
    String pageToken = null;
    do {
      ListAdClientsResponse response = adsense.accounts().adclients().list(accountId)
          .setPageSize(maxPageSize)
          .setPageToken(pageToken)
          .execute();
      List<AdClient> adClients = response.getAdClients();

      if (adClients != null && !adClients.isEmpty()) {
        allAdClients.addAll(adClients);
        for (AdClient adClient : adClients) {
          System.out.printf("Ad client for product \"%s\" with ID \"%s\" was found.\n",
              adClient.getProductCode(), adClient.getName());
          boolean supportsReporting = true;
          if (adClient.getReportingDimensionId() == null
              || adClient.getReportingDimensionId().isEmpty()) {
            supportsReporting = false;
          }
          System.out.printf("\tSupports reporting: %s\n", supportsReporting ? "No" : "Yes");
        }
      } else {
        System.out.println("No ad clients found.");
      }

      pageToken = response.getNextPageToken();
    } while (pageToken != null);

    System.out.println();

    return allAdClients;
  }
}
